gateway/src/apicast/policy_chain.lua

```diff
@@ -96,6 +109,25 @@ function _M:export()
     return chain
 end
 
+--- Freeze the policy chain to prevent modifications.
+-- After calling this method it won't be possible to insert more policies.
+-- @treturn self
+function _M:freeze()
+    self.frozen = true
+    return self
+end
+
+--- Insert a policy into the chain
+-- @tparam Policy policy the policy to be added to the chain
+-- @tparam[opt] int position the position to add the policy to, defaults to last one
+function _M:insert(policy, position)
+    if self.frozen then
+        return nil, 'frozen chain'
```

Comment:
Maybe we should be more specific in the error message. Not sure this will be obvious to contributors.

Reply:
I think the error messages in openresty are pretty simple because people can treat them like error code and match them. I'd probably even go for just `'frozen'` so it is easier to match.

Error codes from [lua-resty-redis](https://github.com/openresty/lua-resty-redis/blob/master/lib/resty/redis.lua):

* not initialized
* not subscribed
* closed
* timeout

If "frozen" is not understandable I could rename it to: immutable, finalized, locked (other suggestions welcome).

Reply:
Fair enough. I was thinking more about clarifying what the error is about to possible contributors rather than making sure it's easy to match.

Reply:
That probably should be addressed by the method documentation describing the error messages. Will do 👍

Reply:
Fixed in https://github.com/3scale/apicast/pull/496/commits/f4be8fb4f04f5588db901b00b97c1965d40b5078 and going to be squashed.